### PR TITLE
Develop, test error

### PR DESCRIPTION
**Pull request by khorkovenko**

### CODE REVIEW

File: store/src/main/java/com/book/store/service/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -51,18 +51,30 @@ public BookDto createBook(BookRequestDto bookDto) {
...
+                .orElseThrow(
+                        () -> new EntityNotFoundException("Book with id "
+                        + id + " not found."));
+
+        existingBook.setTitle(updateBookDto.getTitle());
+        existingBook.setAuthor(updateBookDto.getAuthor());
+        existingBook.setIsbn(updateBookDto.getIsbn());
+        existingBook.setPrice(updateBookDto.getPrice());
+        existingBook.setDescription(updateBookDto.getDescription());
+        existingBook.setCoverImage(updateBookDto.getCoverImage());
```
Comment (Sanderfuck) [inline store/src/main/java/com/book/store/service/impl/BookServiceImpl.java]: Maybe better here use some mapper?

File: store/src/main/java/com/book/store/repo/BookRepository.java [hunk 2]
```diff
@@ -10,8 +10,16 @@
...
+    @Query(
+            "SELECT b FROM Book b JOIN b.categories c WHERE c.id "
+                    + "= :categoryId AND b.deleted = false"
+    )
+    Page<Book> findAllByCategoryIdAndNotDeleted(
+            @Param("categoryId") Long categoryId,
+            Pageable pageable
+    );
+
+    Optional<Book> findByIdAndDeletedFalse(Long id);
```
Comment (Sanderfuck) [inline store/src/main/java/com/book/store/repo/BookRepository.java:23]: Why do you need this methods?

File: store/src/test/java/com/book/store/StoreApplicationTests.java [hunk 3]
```diff
@@ -1,8 +1,16 @@
 package com.book.store;
 
+import org.junit.Test;
+import org.junit.runner.RunWith;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
 
+@RunWith(SpringRunner.class)
```
Comment (Sanderfuck) [inline store/src/test/java/com/book/store/StoreApplicationTests.java]:
```suggestion
```

File: store/src/test/java/com/book/store/StoreApplicationTests.java [hunk 4]
```diff
@@ -1,8 +1,16 @@
...
 @SpringBootTest
-class StoreApplicationTests {
+public class StoreApplicationTests {
```
Comment (Sanderfuck) [inline store/src/test/java/com/book/store/StoreApplicationTests.java]:
```suggestion
class StoreApplicationTests {
```

File: store/src/test/java/com/book/store/controller/BookControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+        Book book = new Book();
+        book.setTitle(title);
+        book.setAuthor(author);
+        book.setIsbn(isbn);
+        book.setPrice(price);
+        book.setDescription(description);
+        book.setCoverImage(coverImage);
+        book.getCategories().add(category);
+        return bookRepository.save(book);
+    }
```
Comment (Sanderfuck) [inline store/src/test/java/com/book/store/controller/BookControllerTest.java]: Private methods must placed after public

File: store/src/test/java/com/book/store/controller/BookControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+    private Book createTestBook(String title, String author, String isbn, BigDecimal price, String description, String coverImage, Category category) {
+        Book book = new Book();
+        book.setTitle(title);
+        book.setAuthor(author);
+        book.setIsbn(isbn);
+        book.setPrice(price);
+        book.setDescription(description);
+        book.setCoverImage(coverImage);
+        book.getCategories().add(category);
+        return bookRepository.save(book);
```
Comment (Sanderfuck) [inline store/src/test/java/com/book/store/controller/BookControllerTest.java]: Just create template of your entity and reuse it

File: store/src/test/java/com/book/store/controller/BookControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+    }
+
+    @Test
+    @DisplayName("getBookById -> WhenBookSaved -> ReturnsCorrectDto")
+    @Transactional
+    @Rollback
+    public void getBookById_WhenBookSaved_ReturnsCorrectDto() throws Exception {
+        Category category = createTestCategory();
+        Book book1 = createTestBook("Book 1", "Author 1", "1234566690007", BigDecimal.valueOf(29.99), "Description 1", "cover1.jpg", category);
```
Comment (Sanderfuck) [inline store/src/test/java/com/book/store/controller/BookControllerTest.java]: Use template here and your code will be looks more clear

File: store/src/test/java/com/book/store/repo/BookRepositoryTests.java [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+        book2 = new Book();
+        book2.setTitle("Book 2");
+        book2.setAuthor("Author 2");
+        book2.setIsbn("9843543210987");
+        book2.setPrice(BigDecimal.valueOf(39.99));
+        book2.setDescription("Description 2");
+        book2.setCoverImage("cover2.jpg");
+        book2.getCategories().add(category);
+        bookRepository.save(book2);
```
Comment (Sanderfuck) [inline store/src/test/java/com/book/store/repo/BookRepositoryTests.java]: You can create it once and reuse in every test what you need

File: store/src/test/java/com/book/store/repo/UserRepositoryTests.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.Rollback;
+import org.springframework.transaction.annotation.Transactional;
+
+@SpringBootTest
+public class UserRepositoryTests {
+    @Autowired
+    private UserRepository userRepository;
+
+    User newUser;
```
Comment (Sanderfuck) [inline store/src/test/java/com/book/store/repo/UserRepositoryTests.java]: why this field not private?

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

File: store/src/main/java/com/book/store/repo/BookRepository.java (hunk 2 above)
Comment (khorkovenko) [reply store/src/main/java/com/book/store/repo/BookRepository.java:23]: To my mind, these both were in the task description

File: store/src/main/java/com/book/store/config/SecurityConfig.java [hunk 10]
```diff
@@ -26,8 +26,6 @@ public class SecurityConfig {
     @Bean
     public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
         http
```
Comment (sokimaaa) [inline store/src/main/java/com/book/store/config/SecurityConfig.java:29]: return back cors\csrf disabled.

**Pull request merged**